context/src/main/java/io/grpc/Deadline.java

```diff
@@ -231,6 +232,35 @@ public int compareTo(Deadline that) {
     return 0;
   }
 
+  @Override
+  public int hashCode() {
+    return Arrays.asList(this.ticker, this.deadlineNanos, this.isExpired()).hashCode();
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (o == this) {
+      return true;
+    }
+    if (!(o instanceof Deadline)) {
+      return false;
+    }
+
+    final Deadline other = (Deadline) o;
+    final Object this$ticker = this.ticker;
```

Comment:
Please avoid using `$` in var names. Especially for `this$ticker` because `this` is a java keyword.

Reply:
I think just use `this.ticker` every time is fine.

Reply:
Will update the names